### DISCUSSION TITLE
Unauthorized

### DISCUSSION THREAD

**Discussion by AdminLLpT** [fattureincloud/api#286] [Q&A] [answered]
Ricevo questo errore quando provo a creare la fattura come mai?

Exception when calling IssuedDocumentsApi->createIssuedDocument: [401] Client error: `POST https://api-v2.fattureincloud.it/c/######/issued_documents` resulted in a `401 Unauthorized` response: {"error":{"message":"Unauthorized."}}

Le autorizzazioni richieste sono:

`$scopes = [Scope::SETTINGS_READ, Scope::ENTITY_CLIENTS_ALL, Scope::ISSUED_DOCUMENTS_INVOICES_ALL];
$url = $oauth->getAuthorizationUrl($scopes, "EXAMPLE_STATE");`

**Comment by valmoz** [comment, marked as answer]
Buongiorno @AdminLLpT 
Prima di tutto, le consiglierei di provare ad effettuare una chiamata [List User Companies](https://developers.fattureincloud.it/api-reference/#get-/user/companies) sempre da codice. Se anch'essa fallisce, allora probabilmente non sta impostando l'access token in maniera corretta.
In tal caso, le suggerirei di controllare che l'access token inizi con il prefisso corretto "a/"

Qualora invece la chiamata List User Companies dovesse essere eseguita con successo, allora probabilmente il problema sta nel company_id utilizzato all'interno dell'URL. Può confrontarlo con quelli restituiti da tale chiamata per verificare che siano quelli corretti.

Le consiglio anche di provare ad effettuare le stesse chiamate con la nostra [API Reference](https://developers.fattureincloud.it/api-reference), impostando il token che sta utilizzando lato codice.

Qualora non dovesse riuscire le suggerirei di contattare il Supporto Clienti presente sulla pagina di Fatture in Cloud per poter procedere con analisi più dettagliate. La nostra Community infatti dovrebbe essere usata per General Guidance e not per Bug Fix (essendo pubblica, non è possibile condividere informazioni sensibili).

A disposizione
Mauro

**Reply by AdminLLpT** [reply]
Grazie, ho risolto cambiando il Redirect URL* nell'applicazione e nella pagina oauth.php
Prima avevo inserito un'altra pagina dove fare il reindirizzamento ma così facendo non si genera l'access token, quindi ho dovuto inserire come Redirect url l'indirizzo della pagina oauth.php e solo così funziona.
